shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -2534,4 +2553,35 @@ public void release() {
          */
         abstract void releaseInternal();
     }
+
+    /**
+     * Convenience class to encapsulate the contents of an attribute update when several should be serialized in the same
+     * WireCommand.
+     */
+    @Data
+    public static final class ConditionalAttributeUpdate {
+        public static final int LENGTH = 4 * Long.BYTES; // UUID (2 longs) + oldValue + newValue
+
+        private final UUID attributeId;
```

Comment:
I think we should likely take this one step further. We should include support for `AttributeUpdateType` as well. Perhaps in the future, we'd want to do Replace/ReplaceIfEquals/Accumulate/etc. 

If we're doing this now, we might as well add another byte to indicate the type.

Reply:
Done, now `ConditionalAttributeUpdate` can take any attribute update type.